Facebook integration

RaykPlatzek opened the pull request:
- All code added as per https://github.com/plataformatec/devise/wiki/OmniAuth%3A-Overview
- devise users controller added
- also added Figaro to make API Keys secure

By adding: "before_action :authenticate_user!, only: :show" to our games controller we will get a link to sign up with Facebook below our normal sign up fields.
However, this link is not working and the following error message comes up:

![image](https://user-images.githubusercontent.com/25449006/27017446-dbdc23bc-4f6b-11e7-807a-9cb1ffd7a0ae.png)

I tried to white-list different URLs on the Facebook Dev account but it still seems to block the request.
Further investigation required....

mscoutermarsh commented:
For testing in dev, I think if you add "localhost" as the url, FB will accept it

RaykPlatzek commented:
@mscoutermarsh we got the facebook login working and added a couple of more features. Might be good to get a second approval. I added an AJAX get request which will check the website every 2 seconds if a new player has joined the game. Once the second player joined the game the board will be populated. Otherwise the board will stay empty and it will say waiting for other players.

mscoutermarsh commented on app/views/games/show.html.erb:
```diff
@@ -11,3 +12,25 @@
+              };
+      }
+
+  });
```
#fancy :D
